### PR TITLE
lint: Switch golang linter to golangci-lint

### PR DESCRIPTION
**Pull request by ganeshmaharaj**
gometalinter is deprecated and will be archived April '19. The
suggestion is to switch to golangci-lint which is apparently 5x faster
than gometalinter.

Partially fixes: github.com/kata-containers/runtime#1377

### CODE REVIEW

**Comment by ganeshmaharaj** [comment]
Jenkins-ci-Power8-ubuntu-16-04-initrd failed with below error.
```
Building remotely on Power8_slave01 (power8-ubuntu-1604 power8_node) in workspace /home/jenkins/workspace/kata-containers-agent-Power8-ubuntu-16-04-PR-initrd
[WS-CLEANUP] Deleting project workspace...
[WS-CLEANUP] Deferred wipeout is disabled by the job configuration...
ERROR: [WS-CLEANUP] Cannot delete workspace: Unable to delete '/home/jenkins/workspace/kata-containers-agent-Power8-ubuntu-16-04-PR-initrd/go/pkg/linux_ppc64le/github.com/kata-containers/tests/vendor/github.com/BurntSushi/toml.a'. Tried 3 times (of a maximum of 3) waiting 0.1 sec between attempts.
ERROR: Cannot delete workspace: Unable to delete '/home/jenkins/workspace/kata-containers-agent-Power8-ubuntu-16-04-PR-initrd/go/pkg/linux_ppc64le/github.com/kata-containers/tests/vendor/github.com/BurntSushi/toml.a'. Tried 3 times (of a maximum of 3) waiting 0.1 sec between attempts.
```

File: channel.go [hunk 1]
```diff
@@ -131,7 +131,7 @@ func (c *serialChannel) wait() error {
 	var events [1]unix.EpollEvent
 
 	fd := c.serialConn.Fd()
-	if fd <= 0 {
+	if fd == 0 {
```
Comment (devimc) [inline channel.go]: is this change needed? :worried:

Comment (sboeuf) [reply channel.go]: Looks safe since https://golang.org/pkg/os/#File.Fd returns a `uintptr`. Don't need to worry about negative values.

Comment (ganeshmaharaj) [reply channel.go]: yup. that was the complaint by the linter.

Comment (devimc) [reply channel.go]: thanks @sboeuf @ganeshmaharaj
